### PR TITLE
adding buoy cli

### PR DESCRIPTION
**Pull request by n3wscott**
**What this PR does, why we need it**:
Bringing in [buoy](https://github.com/n3wscott/buoy) to knative/test-infra.

This adds the buoy cli. See the readme for details.

/lint

### CODE REVIEW

**Comment by n3wscott** [comment]
/hold

Gonna add some tests.

File: buoy/commands/check.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+			gomodFile := args[0]
+
+			var out io.Writer
+			if verbose {
+				out = os.Stdout
+			}
+
+			err := gomod.Check(gomodFile, release, domain, ruleset, out)
+			if errors.Is(err, gomod.DependencyErr) {
+				_, _ = fmt.Fprintln(os.Stderr, err.Error())
```
Comment (vagababov) [inline buoy/commands/check.go:74]:
```suggestion
				fmt.Fprintln(os.Stderr, err.Error())
```
do you really need those?

File: buoy/commands/check.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+				os.Exit(1)
+				return nil
```
Comment (vagababov) [inline buoy/commands/check.go:76]: do you need the `return nil`. One would think go would catch that.

File: buoy/commands/check.go (hunk 1 above)
Comment (n3wscott) [reply buoy/commands/check.go:74]: if you don't you get warnings

**Comment by n3wscott** [comment]
/unhold

I tested check and float, but not via the `cli`. I think it is good enough. If we really wanted to we could run the same tests and e2e and install the cli locally....  maybe a followup

File: buoy/buoy.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	"log"
+
+	"knative.dev/test-infra/buoy/commands"
+)
+
+func main() {
+	if err := commands.New().Execute(); err != nil {
+		log.Fatal("error during command execution: ", err)
```
Comment (vagababov) [inline buoy/buoy.go:27]:
```suggestion
		log.Fatal("Error during command execution: ", err)
```
this is end user message so a complete sentence.

**Review by vagababov — COMMENTED** [review]
Just a nit

**Review by vagababov — COMMENTED** [review]
/lgtm

**Pull request merged**